### DISCUSSION TITLE
Text to screen

### DISCUSSION THREAD

**Discussion by jonnymaserati** [marcomusy/vedo#284] [Q&A] [answered]
Hello there!

Is it possible to print text to the screen rather than onto a plane in the scene?

I wish to print some data to the screen, say in the top right, but would like it static and not to move relative to the camera.

https://user-images.githubusercontent.com/41046859/103279671-995c8480-49ce-11eb-91c9-6ba6d4326140.mp4

I'd like to print data about the end point as it's animated.

Thanks!

**Comment by marcomusy** [comment, marked as answer]
How cool is that! :) 
You have various options, use a "follower" object which is locked to face the camera (you can make any mesh obj become a follower by adding `mesh.followCamera()`) or just overlay some text on the rendering window:
```python
from vedo import *

s = Arrow([0,0,1], [1,1,2])

cam = dict(pos=(4.80, -4.56, 3.32),
           focalPoint=(0.186, 0.0532, 1.05),
           viewup=(-0.230, 0.235, 0.944),
           distance=6.90, clippingRange=(4.75, 9.90))
# (press "C" in rendering window to get the above)

axs = Axes(xrange=(-1,1), yrange=(-1,1), zrange=(0,3), yzGrid=0)

vd=Video()
for i in range(100):
    s.rotateZ(3.6).addShadow(z=0)
    msg1 = Text2D("Some welleng data:\n\Lambda = 10 km",
                 pos="top-left", font="VictorMono", c='w', bg='db', alpha=1)
    msg2 = Text("my |\Psi> vector", font="Theemim", italic=1)
    msg2.scale(.075).pos(s.centerOfMass()+[0.2,0,0]).followCamera()
    plt = show(s, msg1, msg2, axs, camera=cam, resetcam=0, interactive=0)
    vd.addFrame()
    plt.remove([msg1, axs], render=False) # need to remove these manually
vd.close()
interactive()
```
https://user-images.githubusercontent.com/32848391/103285907-96b65b00-49df-11eb-94ef-025ebdb2a0c5.mp4

![ezgif com-gif-maker](https://user-images.githubusercontent.com/32848391/103286269-35db5280-49e0-11eb-9dcc-0efdac8cefa6.gif)

**Reply by jonnymaserati** [reply]
You're a star, especially with that `vd=Video()` part, that saved me a lot of time!

Do you know if any of the fonts are fixed with (all characters the same width)? Think it'd look nicer if the `inc` and `azi` text wasn't jittering so much...

:-)

![wiggly_well](https://user-images.githubusercontent.com/41046859/103296799-274c6580-49f7-11eb-8b7e-699c8157814a.gif)

**Comment by marcomusy** [comment]
Nice!!
Monospaced are "Calco", "Glasgo", "VictorMono", you may also use "\circ" for deg symbol

**Comment by jonnymaserati** [comment]
Awesome... you're a star!

![wiggly_well](https://user-images.githubusercontent.com/41046859/103299065-03d7e980-49fc-11eb-9183-36050dc01c8a.gif)
